checker/jdk/index/src/java/nio/CharBuffer.java

```diff
@@ -0,0 +1,1524 @@
+     *
+     * @param  ob  The object to which this buffer is to be compared
+     *
+     * @return  <tt>true</tt> if, and only if, this buffer is equal to the
+     *           given object
+     */
+    @SuppressWarnings("index:argument.type.incompatible") // #1: i < this.limit() and j < that.limit() as can be seen by the loop's definition
+    public boolean equals(Object ob) {
+        if (this == ob)
+            return true;
+        if (!(ob instanceof CharBuffer))
+            return false;
+        CharBuffer that = (CharBuffer)ob;
+        if (this.remaining() != that.remaining())
+            return false;
+        int p = this.position();
+        for (int i = this.limit() - 1, j = that.limit() - 1; i >= p; i--, j--)
+            if (!equals(this.get(i), that.get(j))) // #1
+                return false;
+        return true;
+    }
+
+    private static boolean equals(char x, char y) {
+
+
+
+        return x == y;
+
+    }
+
+    /**
+     * Compares this buffer to another.
+     *
+     * <p> Two char buffers are compared by comparing their sequences of
+     * remaining elements lexicographically, without regard to the starting
+     * position of each sequence within its corresponding buffer.
+
+
+
+
+
+
+
+
+     * Pairs of {@code char} elements are compared as if by invoking
+     * {@link Character#compare(char,char)}.
+
+     *
+     * <p> A char buffer is not comparable to any other type of object.
+     *
+     * @return  A negative integer, zero, or a positive integer as this buffer
+     *          is less than, equal to, or greater than the given buffer
+     */
+    @SuppressWarnings("index:argument.type.incompatible") // #1: i < this.limit() and j < that.limit() as can be seen by the loop's definition
+    public int compareTo(CharBuffer that) {
+        int n = this.position() + Math.min(this.remaining(), that.remaining());
+        for (int i = this.position(), j = that.position(); i < n; i++, j++) {
+            int cmp = compare(this.get(i), that.get(j)); // #1
+            if (cmp != 0)
+                return cmp;
+        }
+        return this.remaining() - that.remaining();
+    }
+
+    private static int compare(char x, char y) {
+
+
+
+
+
+
+        return Character.compare(x, y);
+
+    }
+
+    // -- Other char stuff --
+
+
+
+    /**
+     * Returns a string containing the characters in this buffer.
+     *
+     * <p> The first character of the resulting string will be the character at
+     * this buffer's position, while the last character will be the character
+     * at index <tt>limit()</tt>&nbsp;-&nbsp;1.  Invoking this method does not
+     * change the buffer's position. </p>
+     *
+     * @return  The specified string
+     */
+    @SuppressWarnings("index:return.type.incompatible") // As in documentation, this function returns a string containing the characters in the buffer
+    public @SameLen("this") String toString() {
```

Comment:
If we're going to annotate this method as `@SameLen("this")`, we should probably also annotate the version above that returns a char array